docs/modules.md

```diff
@@ -56,6 +56,24 @@ To write an Fx module:
    )
    ```
 
+5. Lastly, if you want to keep a constructor hidden to your
+   module (and modules your module includes), you can add an
+   `fx.Private` when providing.
+
+   ```go mdox-exec='region ex/modules/module.go private'
+   var Module = fx.Module("server",
+   	fx.Provide(
+   		New,
+   	),
+   	fx.Provide(
+   		fx.Private,
+   		parseConfig,
+   	),
+   	fx.Invoke(startServer),
+   	fx.Decorate(wrapLogger),
+   )
+   ```
+
```

Comment:
Add some more explanation here about the output of `parseConfig` being only available within the `"server"` module.